array_append.json

```diff
@@ -25,15 +25,20 @@
         },
         {
             "name": "label",
-            "description": "If the given array is a labeled array, a new label for the new value should be given. If not given or `null`, the array index as string is used as the label. If in any case the label exists, a `LabelExists` exception is thrown.",
+            "description": "If the given array is a labeled array, a new label for the new value should be given. If not given or `null`, the array index as number is used as the label. If in any case the label exists, a `LabelExists` exception is thrown.",
```

Comment:
> If not given or `null`, the array index as number is used as the label

not sure what "array index" means in this context. Does it mean something like original length? e.g. `array_append([10, 11], 12)`: length is 2 -> new label = 2

Reply:
Yes, indeed. It's the natural next array index pretty much ;-)

Reply:
makes me wonder: what happens with a non-labeled array and the given label is "out of range"

e.g. `array_append([10, 11], value=20, label=100)`

does that result in `[10, 11, NaN, NaN, ......, NaN, 20]`? or is that invalid

and what if the label is negative?

Reply:
Labels can't be out of range. They behave as if they would be a string, same applies for negative.

But if you pass a non-labeled array and want to add an element with a label, I'm not sure whether we should convert or throw an error?

Reply:
(our labeled arrays pretty much work like PHP arrays 😅)

Reply:
> But if you pass a non-labeled array and want to add an element with a label, I'm not sure whether we should convert or throw an error?

what do you mean with convert? first convert the non-labeled array to labeled array and then use the label as-is?

> (our labeled arrays pretty much work like PHP arrays 😅)

Back in the day when I did PHP development, the frankenstein character of PHP arrays was one of the things I hated most FYI.

Reply:
> what do you mean with convert? first convert the non-labeled array to labeled array and then use the label as-is?

The complexity probably means we should throw an error. 

> he frankenstein character of PHP arrays was one of the things I hated most

I love them and still sometimes miss them lol

Reply:
> But if you pass a non-labeled array and want to add an element with a label, I'm not sure whether we should convert or throw an error?

I'd say throw an error. Implicit conversion of non-labeled to labeled from an append operation doesn't feel right to me